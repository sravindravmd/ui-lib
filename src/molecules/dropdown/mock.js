export const options = [
  { id: 1, name: 'Josh Weir' },
  { id: 2, name: 'Sarah Weir' },
  { id: 3, name: 'Alicia Weir' },
  { id: 4, name: 'Ashutosh Tiwari' },
]
export const dropdownPropsWithTrue = {
  isDefault: true,
  options: [...options],
  titleLabel: 'Field Label',
  value: { id: 0, name: 'Menu Title' },
}
export const dropdownPropsWithFalse = {
  isDefault: false,
  options: [...options],
  titleLabel: 'Field Label',
  value: { id: 0, name: 'Menu Title' },
}
