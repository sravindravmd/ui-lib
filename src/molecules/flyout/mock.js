import React from 'react'
import styled from 'styled-components'
import { theme } from '../../shared'

const StyledFlyoutContent = styled.div`
  box-sizing: border-box;
  font-family: 'Gilroy-Regular';
  width: 200px;

  ul {
    list-style: none;
    padding: 0px;
    margin: 0px;

    li {
      a {
        font-size: 13px;
        line-height: 16px;
        letter-spacing: 0.02em;
        text-decoration: none;
        color: ${theme.palette.brand.primary.charcoal};
        padding: 10px 23px;
        display: block;
        box-sizing: border-box;
        transition: all 0.2s ease-in-out;

        &:hover {
          background-color: #f4f4f4;
        }
      }
    }
  }
`

export const FlyoutContent = () => {
  return (
    <StyledFlyoutContent>
      <ul>
        <li>
          <a href=''>Item</a>
        </li>
        <li>
          <a href=''>Item</a>
        </li>
        <li>
          <a href=''>Item</a>
        </li>
        <li>
          <a href=''>Item</a>
        </li>
      </ul>
    </StyledFlyoutContent>
  )
}
