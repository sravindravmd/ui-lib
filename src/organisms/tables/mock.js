export const data = {
  tableData: [
    {
      id: 1,
      orderId: '01283884',
      created: '5 min ago',
      customerName: 'Annebelle Sinh',
      total: '$130',
      profit: '$70 / 80%',
      status: {
        value: { id: 1, name: 'In Progress' },
        options: [
          { id: 1, name: 'In Progress' },
          { id: 2, name: 'Ready to Ship' },
          { id: 3, name: 'Out for Delivery' },
        ],
      },
      notes: 'ABCD',
    },
    {
      id: 2,
      orderId: '01283885',
      created: '30 min ago',
      customerName: 'Rayna Baptista',
      total: '$40',
      profit: '$10 / 20%',
      status: {
        value: { id: 3, name: 'Out for Delivery' },
        options: [
          { id: 1, name: 'In Progress' },
          { id: 2, name: 'Ready to Ship' },
          { id: 3, name: 'Out for Delivery' },
        ],
      },
      hasChild: true,
      childROW: [
        {
          id: 2,
          orderId: '',
          created: '42 min ago',
          customerName: 'Charlie Workman',
          total: '$40',
          profit: '$10 / 20%',
          status: {
            value: { id: 2, name: 'Ready to Ship' },
            options: [
              { id: 1, name: 'In Progress' },
              { id: 2, name: 'Ready to Ship' },
              { id: 3, name: 'Alicia Weir' },
            ],
          },
          notes: 'ABCD',
        },
      ],
      notes: 'ABCD',
    },
    {
      id: 3,
      orderId: '01283886',
      created: '42 min ago',
      customerName: 'Charlie Workman',
      total: '$40',
      profit: '$10 / 20%',
      status: {
        value: { id: 2, name: 'Ready to Ship' },
        options: [
          { id: 1, name: 'In Progress' },
          { id: 2, name: 'Ready to Ship' },
          { id: 3, name: 'Alicia Weir' },
        ],
      },
      notes: 'ABCD',
    },
  ],
  headerElement: [
    { headingLabel: '#', headingKey: '', isSortable: false },
    { headingLabel: 'Order ID', headingKey: 'orderId', isSortable: true },
    { headingLabel: 'Created', headingKey: 'created', isSortable: true },
    {
      headingLabel: 'Customer Name',
      headingKey: 'customerName',
      isSortable: true,
    },
    { headingLabel: 'Total', headingKey: 'total', isSortable: true },
    { headingLabel: 'Profit', headingKey: 'profit', isSortable: true },
    {
      headingLabel: 'Status',
      headingKey: 'status',
      isSortable: true,
    },
    { headingLabel: 'Notes', headingKey: 'notes', isSortable: true },
  ],
  isTableSortable: false,
  className: 'test-class',
}
