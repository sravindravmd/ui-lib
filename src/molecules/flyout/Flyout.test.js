import React from 'react'
import { render, cleanup } from '@testing-library/react'
import 'regenerator-runtime/runtime'
import { Flyout } from './'
import { FlyoutContent } from './mock'

afterEach(cleanup)

const prop = {
  show: true,
}

describe('<Flyout/>', () => {
  it('renders', () => {
    const { container } = render(
      <Flyout {...prop}>
        <FlyoutContent />
      </Flyout>
    )
    expect(container).toMatchSnapshot()
  })
})
