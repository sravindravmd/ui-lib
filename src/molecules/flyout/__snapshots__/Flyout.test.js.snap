// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flyout/> renders 1`] = `
<div>
  <div
    class="sc-bdfBwQ bhyxXU active"
  >
    <div
      class="content"
    >
      <div
        class="content-inner"
      >
        <div
          class="sc-gsTCUz kTbkcH"
        >
          <ul>
            <li>
              <a
                href=""
              >
                Item
              </a>
            </li>
            <li>
              <a
                href=""
              >
                Item
              </a>
            </li>
            <li>
              <a
                href=""
              >
                Item
              </a>
            </li>
            <li>
              <a
                href=""
              >
                Item
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
